test/integration/apiserver.go

```diff
@@ -197,22 +197,170 @@ func newAPIServer(f *testutil.Framework) func(t *testing.T) {
 			testutil.LogObject(t, userinfo)
 		}
 
-		for name, testFn := range map[string]func(ctx context.Context, conn *grpc.ClientConn, managementClient *testutil.RecordingClient) func(t *testing.T){
+		for name, testFn := range map[string]func(ctx context.Context, conn *grpc.ClientConn, managementClient *testutil.RecordingClient, f *testutil.Framework) func(t *testing.T){
 			"offering-service": offeringService,
 			"region-service":   regionService,
 			"provider-service": providerService,
+			"instance-service": instanceService,
 		} {
 			name := name
 			testFn := testFn
 
 			t.Run(name, func(t *testing.T) {
 				t.Parallel()
-				testFn(ctx, conn, managementClient)(t)
+				testFn(ctx, conn, managementClient, f)(t)
 			})
 		}
 	}
 }
 
+func instanceService(ctx context.Context, conn *grpc.ClientConn, managementClient *testutil.RecordingClient, f *testutil.Framework) func(t *testing.T) {
+	return func(t *testing.T) {
+		serviceClient, err := f.ServiceClient(t)
+		require.NoError(t, err, "creating service client")
+		ctx, cancel := context.WithCancel(context.Background())
+		t.Cleanup(cancel)
+		managementClient, err := f.ManagementClient(t)
```

Comment:
I don't think you need the managementClient and create a new context in the function, you can just use them by passing from the function, right?

Reply:
you are right

Reply:
done